ote/src/clj/ote/services/transit_visualization.clj

```diff
@@ -18,32 +18,42 @@
 
 (defn- parse-stops [stops]
   (mapv (fn [stop]
-          (let [[lat lon stop-name] (str/split stop #",")]
+          (let [[lat lon stop-name trip-id headsign] (str/split stop #",")]
             {:lat (Double/parseDouble lat)
              :lon (Double/parseDouble lon)
-             :name stop-name}))
+             :stop-name stop-name
+             :trip-id trip-id
+             :headsign headsign}))
         (str/split stops #"\|\|")))
 
-(defn route-line-features [trips]
-  (mapcat (fn [{:keys [route-line departures stops] :as foo}]
-            (let [all-stops (parse-stops stops)
-                  first-stop (first all-stops)
-                  last-stop (last all-stops)]
-              (vec (into
-                    #{{:type "Feature"
-                       :properties {:departures (mapv time/format-interval-as-time (.getArray departures))
-                                    :routename (str (:name first-stop) " \u2192 " (:name last-stop))}
-                       :geometry (cheshire/decode route-line keyword)}}
-                    (map (fn [stop]
-                           (let [[lon lat name] (str/split stop #",")]
-                             {:type "Point"
-                              :coordinates [(Double/parseDouble lon)
-                                            (Double/parseDouble lat)]
-                              :properties {"name" name
-                                           "trip-name" (str (:name first-stop) " \u2192 " (:name last-stop))}})))
-                    (when-not (str/blank? stops)
-                      (str/split stops #"\|\|"))))))
-          trips))
+(defn trip-lines [trips]
```

Comment:
Food for thought: Would the long function with nested iterations become more readable, by e.g. splitting the loops into functions or naming the anonymous functions?

Reply:
You are correct. This is kind of hard to understand what is happening here and why. I try to add comments